ehsm_kms_service/ehsm_kms_server.js

```diff
@@ -0,0 +1,261 @@
+      return;
+    }
+  }
+  next();
+}
+/**
+ * sign 
+ */
+
+app.use(_checkSign);
+/**
+ * payload 
+ */
+app.use(_checkPayload);
+
+/**
+ * ehsm_napi function object
+ */ 
+const ehsm_napi = ffi.Library('./libehsmnapi',{
+  /**
+    NAPI_CreateKey
+    Description:
+    Create a customer master key with the following metadatas
+    .keyspec;
+        -EH_AES_GCM_128,
+        -EH_AES_GCM_256,
+        -EH_RSA_2048,
+        -EH_RSA_3072,
+        -EH_EC_P256,
+        -EH_EC_P512,
+        -EH_EC_SM2,
+        -EH_SM4,
+    .origin;
+        -EH_INTERNAL_KEY (generated from the eHSM inside)
+        -EXTERNAL_KEY (generated by the customer and want to import into the eHSM),
+    .purpose;
+        -ENCRYPT_DECRYPT,
+        -SIGN_VERIFY,
+    .apiversion;
+        -Reserved
+    .descrption;
+        -Reserved
+    .createdate
+        -Reserved
+    Note: the CMK will be wrapped by the DK(DomainKey)
+
+    params 
+     - keyspec: int
+     - origin: int
+    
+    return json
+     {
+       code: int,
+       message: string,
+       result: {
+         cmk_base64 : string,
+       }
+    }
+   */
+  'NAPI_CreateKey': ['string',['int','int']],
+  /**
+    NAPI_Encrypt 
+    Description:
+     Encrypt an arbitrary set of bytes using the CMK.(only support symmetric types)
+
+    params
+     - cmk_base64: string
+     - plaintext: string
+     - aad: string
+    
+    return json
+     {
+       code: int,
+       message: string,
+       result: {
+         cipherText_base64 : string,
+       }
+    }
+   */
+  'NAPI_Encrypt': ['string',['string','string', 'string']],
+  
+  /**
+    NAPI_Decrypt 
+    Description:
+      Decrypts ciphertext using the CMK.(only support symmetric types)
+
+    params
+     - cmk_base64: string
+     - ciphertext: string
+     - aad: string
+    
+    return json
+     {
+       code: int,
+       message: string,
+       result: {
+         plaintext_base64 : string,
+       }
+    }
+   */
+  'NAPI_Decrypt': ['string',['string','string', 'string']],
+  /**
+    NAPI_GenerateDataKey 
+    Description:
+    Generates a random data key that is used to locally encrypt data.
+    the datakey will be wrapped by the specified CMK(only support asymmetric keyspec),
+    and it will return the plaintext and ciphertext of the data key.
+    You can use the plaintext of the data key to locally encrypt your data without using KMS
+    and store the encrypted data together with the ciphertext of the data key, then clear the
+    plaintext data from memory as soon as possible.
+    when you want to obtain the plaintext of datakey again, you can call the Decrypt with the
+    cmk to get the plaintext data.
+
+    params 
+     - cmk_base64: string
+     - aad: string
+     
+    return json
+     {
+       code: int,
+       message: string,
+       result: {
+         plaintext_base64 : string,
+         cipherText_base64 : string,
+       }
+    }
+   */
+  'NAPI_GenerateDataKey': ['string',['string','string']],
+
```

Comment:
the backend API NAPI_GenerateDataKey lost a parameter "keylen"

Reply:
It will be fix.